### PR TITLE
[SCHEMA] Very minor tune ups to the test_rule_objects test while trying to grasp its function

### PR DESCRIPTION
**Pull request by yarikoptic**

### CODE REVIEW

File: tools/schemacode/bidsschematools/data/tests/test_rules.py [hunk 1]
```diff
@@ -83,8 +83,8 @@ def test_rule_objects(schema_obj):
                 if object_type in ["extensions", "suffixes"]:
                     # Some object types are referenced via their "value" fields in the rules
                     object_values = [
-                        schema_obj["objects"][object_type][k]["value"]
-                        for k in schema_obj["objects"][object_type].keys()
+                        value["value"]
+                        for value in schema_obj["objects"][object_type].values()
```
Comment (yarikoptic) [inline tools/schemacode/bidsschematools/data/tests/test_rules.py:87]: simpler == easier to read and understand

File: tools/schemacode/bidsschematools/data/tests/test_rules.py [hunk 2]
```diff
@@ -100,7 +100,7 @@ def test_rule_objects(schema_obj):
 
     if not_found:
         not_found_string = "\n".join([f"{'.'.join(path)} == {val}" for path, val in not_found])
-        raise ValueError(not_found_string)
+        raise AssertionError(f"Undefined objects found in rules: {not_found_string}"
```
Comment (yarikoptic) [inline tools/schemacode/bidsschematools/data/tests/test_rules.py:103]: it is a test, so raising AssertionError is expected

Comment (effigies) [inline tools/schemacode/bidsschematools/data/tests/test_rules.py:103]:
```suggestion
        raise AssertionError(f"Undefined objects found in rules: {not_found_string}")
```

**Review by effigies — APPROVED** [review]

File: tools/schemacode/bidsschematools/data/tests/test_rules.py [hunk 3]
```diff
@@ -100,7 +100,7 @@ def test_rule_objects(schema_obj):
 
     if not_found:
         not_found_string = "\n".join([f"{'.'.join(path)} == {val}" for path, val in not_found])
-        raise AssertionError(f"Undefined objects found in rules: {not_found_string}"
+        raise AssertionError(f"Undefined objects found in rules: {not_found_string}")
```
Comment (yarikoptic) [inline tools/schemacode/bidsschematools/data/tests/test_rules.py:102]: I could have sworn I have tested it -- must be some hand tremble removed it last moment ;)

**Review by TheChymera — APPROVED** [review]

**Pull request merged**